### PR TITLE
Erlang 17 compatibility

### PR DESCRIPTION
**Pull request by joedevivo**
Ran into an issue with erlang changing the default way it encodes things. I think it might default binaries to UTF-16, but I can't honestly tell. Something bigger than UTF-8 for sure. Making sure the input was UTF-8 for these tests ensures that the functionality of the nif is still valid. 

An outstanding question is "does the default encoding of binary strings in erlang 17 change the overall functionality of chef_reindex?" 

ping @chef/lob and @oferrigni

### CODE REVIEW

**Comment by joedevivo** [comment]
Here's a cool link to "Using Unicode in Erlang" http://www.erlang.org/doc/apps/stdlib/unicode_usage.html

File: test/fast_xs_tests.erl [hunk 1]
```diff
@@ -27,17 +25,18 @@ predefined_test() ->
...
     ok = io:setopts([{encoding, unicode}]),
     ?assertEqual(<<"*">>, fast_xs:escape(<<"\x00">>)),               % null
     ?assertEqual(<<"*">>, fast_xs:escape(<<"\x0C">>)),               % form feed
-    ?assertEqual(<<"*">>, fast_xs:escape(<<65536>>)).           
+    ?assertEqual(<<"*">>, fast_xs:escape(<<65536>>)).
 
 iso_8859_misc_test() ->
-    ?assertEqual(<<"&#239;&#191;&#189;">>, fast_xs:escape(<<"�">>)),
-    ?assertEqual(<<"name__=__Microsoft&#194;&#174;">>, fast_xs:escape(<<"name__=__Microsoft®">>)),
+    io:format("printf~n"),
```
Comment (oferrigni) [inline test/fast_xs_tests.erl]: printf?

File: include/chef_types.hrl [hunk 2]
```diff
@@ -0,0 +1,222 @@
+%% Copyright 2012 Opscode, Inc. All Rights Reserved.
```
Comment (oferrigni) [inline include/chef_types.hrl]: How are you going to make sure this stays up to date?

File: src/chef_ez_reindex_direct.erl [hunk 3]
```diff
@@ -18,7 +18,11 @@
 
 -module(chef_ez_reindex_direct).
 
--include_lib("oc_erchef/include/chef_types.hrl").
+-ifdef(namespaced_types).
+-type dict() :: dict:dict().
+-endif.
```
Comment (oferrigni) [inline src/chef_ez_reindex_direct.erl]: This is awesome!

File: test/fast_xs_tests.erl (hunk 1 above)
Comment (joedevivo) [inline test/fast_xs_tests.erl]: will remove that

**Comment by joedevivo** [comment]
ok, review on this is ready to resume

**Comment by stevendanna** [comment]
:+1:

**Pull request merged**